src/App.tsx

```diff
@@ -1,27 +1,47 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const handleLoadData = async (request: () => Promise<Good[]>) => {
+    await request()
+      .then(data => setGoods(data))
+      .catch(() => new Error('This request is not handle'));
+  };
```

Comment:
amazing solution❤️

Reply:
FYI: it's better to handle an error and set it's message in a separated state and show on the UI so user would know the reason of a problem